ext/standard/string.c

```diff
@@ -5653,6 +5655,98 @@ PHP_FUNCTION(substr_compare)
 }
 /* }}} */
 
+/* {{{ */
+static zend_string *php_utf8_encode(const char *s, size_t len)
+{
+	size_t pos = len;
+	zend_string *str;
+	unsigned char c;
+
+	str = zend_string_safe_alloc(len, 4, 0, 0);
```

Comment:
Good catch. The original function was (needlessly!) more generic and supported encoding 3- and 4-byte characters, which I stripped out, but I missed this.